runtime/j9vm/j8vmi.c

```diff
@@ -99,3 +99,114 @@ JVM_GetTemporaryDirectory(JNIEnv *env)
 	return result;
 }
 
+
+/**
+ * Copies memory from one place to another, endian flipping the data.
+ *
+ * @param [in] env Pointer to JNI environment
+ * @param [in] srcObj Source primitive array (NULL means srcOffset represents native memory)
+ * @param [in] srcOffset Offset in source array / address in native memory
+ * @param [in] dstObj Destination primitive array (NULL means dstOffset represents native memory)
+ * @param [in] dstOffset Offset in destination array / address in native memory
+ * @param [in] size Number of bytes to copy
+ * @param [in] elemSize Size of elements to copy and flip
+ *
+ * elemSize = 1 means a straight byte copy
+ * elemSize = 2 means byte order 1,2 becomes 2,1
+ * elemSize = 4 means byte order 1,2,3,4 becomes 4,3,2,1
+ * elemSize = 8 means byte order 1,2,3,4,5,6,7,8 becomes 8,7,6,5,4,3,2,1
+ */
+void JNICALL
+JVM_CopySwapMemory(JNIEnv *env, jobject srcObj, jlong srcOffset, jobject dstObj, jlong dstOffset, jlong size, jlong elemSize)
+{
+	void *srcBytes = NULL;
+	void *dstBytes = NULL;
+	U_8 *srcAddr = NULL;
+	U_8 *dstAddr = NULL;
+	/* Ensure that size is a non-negative multiple of elemSize */
+	Assert_SC_true(size >= 0);
+	Assert_SC_true(0 == (size & elemSize));
```

Comment:
I think you meant `0 == (size % elemSize)`.

Reply:
I did, and noticed it in a review with Dan and forgot to fix...

Reply:
Fixed.